engine/script/src/script_timer.cpp

```diff
@@ -84,21 +75,22 @@ namespace dmScript
         uint32_t        m_IsAlive : 1;
     };
 
-    #define INVALID_TIMER_LOOKUP_INDEX  0xffffu
     #define INITIAL_TIMER_CAPACITY      8u
-    #define MAX_TIMER_CAPACITY          65000u  // Needs to be less that 65535 since 65535 is reserved for invalid index
     #define TIMER_CAPACITY_GROWTH       16u
 
     struct TimerWorld
     {
-        dmArray<Timer>                      m_Timers;
-        dmArray<uint16_t>                   m_IndexLookup;
-        dmIndexPool<uint16_t>               m_IndexPool;
-        uint16_t                            m_Version;   // Incremented to avoid collisions each time we push timer indexes back to the m_IndexPool
-        uint16_t                            m_InUpdate : 1;
+        dmObjectPool<Timer> m_Timers;
+        dmSet<uint16_t>     m_Instances; // the pool indices
+        dmArray<uint16_t>   m_ScratchBuffer; // When doing operations on the timer instances, we need a copy to avoid self modification of m_Instances
```

Comment:
The old implementation was essentially doing the same thing but more "manually".
Let's use our ObjectPool as it was intended.